pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -188,3 +351,55 @@ func getInfraMachineTemplateFromProvider(platform configv1.PlatformType) (client
 		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
+
+// updateMAPIMachineSetSpecFromCAPI updates the MAPI MachineSet spec based on the CAPI MachineSet.
+func updateMAPIMachineSetSpecFromCAPI(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet, newProviderSpec machinev1beta1.ProviderSpec) {
```

Comment:
Hmm, how much of this is needed? I expected most of this to be handled inside the conversion library 🤔 

I think there are a handful of things that need to be handled at a higher level, but not sure about these ones.

I think primarily the changes to be handled at a higher level are finalizers and owner references, so this may not be required at all